### ISSUE TITLE
Dragonriding speed?

### ISSUE THREAD

**Issue by Paperiritari** [Arxareon/MovementSpeed#1]
Hi! First of all this addon works very well, thank you for making this! However it doesn't show dragonriding speed, is it even possible to access this information?

**Comment by Arxareon** [comment]
_**Post on CurseForge addressing Dragonriding**: https://www.curseforge.com/wow/addons/movement-speed?comment=35_

_TLDR:_ Sadly, Blizzard doesn't provide a way for addons to access this info while Dragonriding currently. However, I've been working on some features that would other give you useful info which should work while Dragonriding as well.

Features such as calculating the current actual travel speed as opposed to just your character's movement speed (for example, if you're walking into a wall or fly straight up, you will travel forward at/near 0 yards/s while your character's speed is technically still the same, however many yards/sec it was).
And so, the testing I've done with Dragonriding is promising and this feature mentioned will support it. Once I can develop it, figure out the right way to implement & support it, polish it some, you'll be able to see your travel speed, meaning how fast you travel horizontally across the map at any given time. It won't unfortunately be able to tell you your actual flying speed while you dive from a mountain for instance, but it should provide some useful info regardless as to how fast you advance.
I've been focusing on getting this off the ground asap (that's why I'm not even close to being 70/done with the Campaign yet :P) so expect something soon. I'll probably release these features in steps to get critical stuff (like some king of speed during Dragonriding) out the door faster.

**Comment by sc0ttkclark** [comment]
Unfortunate that this info isn't available to add-ons but that's a novel idea to work around that, wish you all the best in your efforts here.

**Comment by Arxareon** [comment]
The new update has a new Travel Speed feature which shows estimated speed values* even during Dragonriding.
[Movement Speed 2.4](https://github.com/Arxareon/MovementSpeed/releases/tag/2.4)

I will close this issue for now. When Blizzard adds API support for player speed state values during Dragonriding, I will add support for it. Any problems & feedback regarding the new Travel Speed feature can be posted as its own separate Issue here on GitHub.

*_The speed at which your character is moving through the map horizontally as your coordinates change._

**Comment by Dairyman** [comment]
FYI: [Dragonriding speed](https://wowpedia.fandom.com/wiki/API_C_PlayerInfo.GetGlidingInfo)

**Comment by Arxareon** [comment]
> FYI: [Dragonriding speed](https://wowpedia.fandom.com/wiki/API_C_PlayerInfo.GetGlidingInfo)

Very nice!
I actually asked for this but completely missed that they added it back in 10.0.5, apparently. I'll get to implementing it as soon as I can!

Thanks for the heads up!

**Comment by gweber68** [comment]
Hi Arxareon, are there any updates to share on progress for getting the new API that accesses player movement & gliding speed integrated into your addon?  I'd love to contribute if you need a hand.

**Comment by Arxareon** [comment]
**Update:**

I had a lot of work I was busy with this fall - still do - but, I've been working on a bigger update I'm hoping to release it next week. It will have a full implementation of the Dragonriding API for precise player speeds.

**Comment by Pixelatex** [comment]
Any updates yet on that new version?

**Comment by Arxareon** [comment]
Well, a lot has happened that burned my time away but now it actually is coming together. I'll release a test version very soon, maybe even this week if things go well.

Thank you for understanding and all the patience! 💖

**Comment by Arxareon** [comment]
Good news, the test version is here (I'll soon upload it as a beta release to CurseForge as well)! https://github.com/Arxareon/MovementSpeed/releases/tag/3.0_Test_A
Please refer to the known issues list in the notes.

I will close this issue as solved in the next update, any feedback is welcome in this issue thread opened specifically for testing: https://github.com/Arxareon/MovementSpeed/issues/14

Sorry I couldn't fit it in last week.
